src/whisper_at_train/run_combined_training.sh

```
@@ -4,23 +4,34 @@
-set -x
-# comment this line if not running on sls cluster
-#. /data/sls/scratch/share-201907/slstoolchainrc
-source /home/taemyung_heo/.cache/pypoetry/virtualenvs/whisper-at-z6hdRBdT-py3.10/bin/activate
-export TORCH_HOME=../../pretrained_models
+set -e  # Exit on any error
+set -x  # Print commands for debugging
+
+# Activate poetry environment if available, otherwise use system python
+if command -v poetry &> /dev/null; then
+    echo "Using poetry environment..."
+    source $(poetry env info --path)/bin/activate
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)